python/ray/client_builder.py

```diff
@@ -57,7 +65,95 @@ def connect(self) -> ClientInfo:
 
 
 class _LocalClientBuilder(ClientBuilder):
-    pass
+    """
+    Connects/or Creates to a local Ray Cluster with Ray Client.
+    """
+
+    def __init__(self, address: Optional[str]) -> None:
+        super().__init__(address)
+        # If local, we should always create a new cluster.
+        self.create_cluster_without_searching = (address == "local")
+        self._init_args = {}
+        self._server = None
+
+    def resources(
+            self,
+            *,
+            num_cpus: Optional[int] = None,
+            num_gpus: Optional[int] = None,
+            resources: Optional[Dict[str, str]] = None,
+            object_store_memory: Optional[int] = None) -> "ClientBuilder":
+        self._init_args["num_cpus"] = num_cpus
+        self._init_args["num_gpus"] = num_gpus
+        self._init_args["resources"] = resources
+        self._init_args["object_store_memory"] = object_store_memory
+        return self
+
+    def dashboard(self,
+                  *,
+                  include: bool = True,
+                  host: str = "127.0.0.1",
+                  port: Optional[int] = None) -> "ClientBuilder":
+        assert self.create_cluster_without_searching
+        self._init_args["include_dashboard"] = include
+        self._init_args["dashboard_host"] = host
+        self._init_args["dashboard_port"] = port
+        return self
+
+    def _create_local_cluster(self):
+        def ray_connect_handler(job_config=None):
+            with disable_client_hook():
+                if not ray.is_initialized():
+                    ray.init(job_config=job_config, **self._init_args)
+
+        port, _ = get_unused_port()
+        self.address = f"localhost:{port}"
+        logger.info(f"Creating a server on {self.address}")
+        self._server = client_server.serve(self.address, ray_connect_handler)
+        atexit.register(ray.shutdown, True)
```

Comment:
We can't use the client server in local cluster mode right? This will be a severe performance regression. Instead we should use the faster ray.init() path here.